Associated services only service & solution selection

magnusjohn-bjss opened the pull request.

magnusjohn-bjss committed:
Associated services only service & solution selection

bjss-colin commented on src/NHSD.GPIT.BuyingCatalogue.Services/Orders/SupplierService.cs:
```diff
@@ -61,6 +61,17 @@ public Task<List<Supplier>> GetAllSuppliersFromBuyingCatalogue()
 
+        public Task<List<Supplier>> GetAllSuppliersWithAssociatedServices()
+        {
+            return dbContext.Suppliers
```
It's worth pinning an `AsNoTracking` on this since it's used in a read-only scenario

bjss-colin commented on src/NHSD.GPIT.BuyingCatalogue.Services/Solutions/SolutionsService.cs:
```diff
@@ -414,6 +414,23 @@ public Task<List<CatalogueItem>> GetSupplierSolutions(int? supplierId)
 
+        public async Task<List<CatalogueItem>> GetSupplierSolutionsWithAssociatedServices(int? supplierId)
+        {
+            return await dbContext.CatalogueItems
```
It's worth pinning an `AsNoTracking` on this since it's used in a read-only scenario

bjss-colin commented on src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Order/Views/Shared/SelectSolution.cshtml:
```diff
@@ -26,43 +26,69 @@
-                    }
-                </nhs-radio-button-container>
-            </nhs-fieldset-form-label>
+            @if (!Model.AssociatedServicesOnly 
```
`if` condition can be shortened to `if (Model.CatalogueSolutions.Count > 1)`, unless everything after the OR needs to be wrapped in parenthesis?

magnusjohn-bjss replied:
&& takes precedence over ||, so the parentheses are implicit (i.e. `!a || a && b` is equivalent to `!a || (a && b)`). want some added in anyway to prevent confusion? vs flags them as unnecessary is the only thing

bjss-colin replied:
That's fine, I couldn't quite remember the order and/or precedence between the two.

bjss-colin reviewed (changes requested):
Looks good, some minor suggestions added

bjss-colin committed:
Review changes

Ensure suppliers are only shown if they have associated services that are referenced by any published solutions

Content updates to the Select Supplier page

bjss-colin committed:
Merge branch 'develop' into feature/20478-select-supplier-and-solutions
